lib/iris/analysis/trajectory.py

```diff
@@ -272,112 +279,115 @@ def interpolate(cube, sample_points, method=None):
         cache = {}
         column_indexes = _nearest_neighbour_indices_ndcoords(
             cube, sample_points, cache=cache)
+        if index_only:
+            return column_indexes
+        else:
```

Comment:
I would suggest you simply omit the else: statement. If index_only is True the function will return and further operation will cease, otherwise it will carry on. It makes the changes look a lot smaller as the indentation needn't change thereafter.